### PR TITLE
Update new post/event email, and refactor related notification logic

### PR DESCRIPTION
**Pull request by s-cheng**
The goal of this PR is to improve the emails we send to notify users of new events. Included here are:
1. Combining all notifications related to **publishing** a new post/event into the `postsNewNotifications()` function (publishing only happens once the post is both approved and not a draft, not necessarily when the post is created)
2. Fixing the logic inside `postsNewNotifications()` so that it sends the proper notification type per user, and only sends one notification per post per user
3. Using the same email template for the "new event in your group" and "new event in your radius" emails
4. Writing tests for `postsNewNotifications()` (which accounts for most of the changes in this PR)
5. Some design updates to the actual "new post" email

This PR is large enough already so I decided to stop here. For the next PR, I'll be cleaning up the "event updated" notifications, and possibly spending more time updating the design of the "new post" email.

---
Here is the updated "new post" email. I just updated the title text font, cleaned up some spacing, and added color to the post metadata section.
Screen Shot 2021-12-27 at 4 00 32 PM

---
Here is the "new event" version. I added the group name, whether it's an online event, and farther down I removed the "Other Recommended Posts" section and included the reason for why they are receiving the email.
Screen Shot 2021-12-27 at 4 00 58 PM

### CODE REVIEW

File: packages/lesswrong/server/callbacks/postCallbacks.ts [hunk 1]
```diff
@@ -21,12 +21,6 @@ getCollectionHooks("Posts").updateBefore.add(function PostsEditRunPostUndraftedS
   return data;
 });
 
-getCollectionHooks("Posts").editAsync.add(function PostsEditRunPostUndraftedAsyncCallbacks (newPost, oldPost) {
-  if (!newPost.draft && oldPost.draft) {
-    void postsUndraftNotification(newPost);
-  }
-});
```
Comment (s-cheng) [inline packages/lesswrong/server/callbacks/postCallbacks.ts:28]: This should already be handled by `updatedPostMaybeTriggerReview` below

File: packages/lesswrong/server/callbacks/postCallbacks.ts [hunk 2]
```diff
@@ -168,14 +162,18 @@ getCollectionHooks("Posts").newAfter.add(async (document: DbPost) => {
...
-  if (!newPost.draft && oldPost.draft) {
+  if (oldPost.draft) {
     await newDocumentMaybeTriggerReview(newPost)
+  }
+  
+  // if the post author is already approved and the post is getting undrafted,
+  // or the post author is getting approved,
+  // then we consider this "publishing" the post
+  if ((oldPost.draft && !newPost.authorIsUnreviewed) || (oldPost.authorIsUnreviewed && !newPost.authorIsUnreviewed)) {
     await postPublishedCallback.runCallbacksAsync([newPost]);
```
Comment (s-cheng) [inline packages/lesswrong/server/callbacks/postCallbacks.ts:177]: I refactored this to make sure `postsNewNotifications()` couldn't be triggered twice by one edit

File: packages/lesswrong/server/notificationCallbacks.tests.ts [hunk 3]
```diff
@@ -1,41 +1,112 @@
...
-// import { getSubscribedUsers, postsNewNotifications } from "./notificationCallbacks";
+import { createDummyPost, createDummyLocalgroup } from "../testing/utils";
+import { postsNewNotifications } from "./notificationCallbacks";
+import { createNotifications } from "./notificationCallbacksHelpers";
 
 testStartup()
 
-jest.mock('./notificationCallbacks', () => {
-  const originalModule = jest.requireActual('./notificationCallbacks');
+jest.mock('./notificationCallbacksHelpers', () => {
```
Comment (s-cheng) [inline packages/lesswrong/server/notificationCallbacks.tests.ts:8]: In order to mock some functions to test `postsNewNotifications()`, I had to move them to a separate module

File: packages/lesswrong/server/notificationCallbacks.tsx [hunk 4]
```diff
@@ -293,66 +67,80 @@ getCollectionHooks("Posts").editAsync.add(function PostsEditRunPostApprovedAsync
   }
 });
 
-export async function postsUndraftNotification(post: DbPost) {
-  //eslint-disable-next-line no-console
-  console.info("Post undrafted, creating notifications");
-
-  await postsNewNotifications(post);
-}
-
```
Comment (s-cheng) [inline packages/lesswrong/server/notificationCallbacks.tsx:302]: This was no longer used so I removed it

File: packages/lesswrong/server/notificationCallbacks.tsx [hunk 5]
```diff
@@ -293,66 +67,80 @@ getCollectionHooks("Posts").editAsync.add(function PostsEditRunPostApprovedAsync
...
 function postIsPublic (post: DbPost) {
   return !post.draft && post.status === postStatuses.STATUS_APPROVED
 }
 
 // Export for testing
-/** Add new post notification callback on post submit */
+/** create notifications for a new post being published */
 export async function postsNewNotifications (post: DbPost) {
```
Comment (s-cheng) [inline packages/lesswrong/server/notificationCallbacks.tsx:76]: This is the only function that changed in this file - the rest were moved to the new file

File: packages/lesswrong/server/notificationCallbacksHelpers.tsx [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+  type: string,
+  potentiallyDefaultSubscribedUserIds?: null|Array<string>,
+  userIsDefaultSubscribed?: null|((u:DbUser)=>boolean),
+}) {
+  if (!documentId) {
+    return [];
+  }
+  
+  const subscriptions = await Subscriptions.find({documentId, type, collectionName, deleted: false, state: 'subscribed'}).fetch()
+  const explicitlySubscribedUserIds = _.pluck(subscriptions, 'userId')
```
Comment (Xodarap) [inline packages/lesswrong/server/notificationCallbacksHelpers.tsx:57]: I think you should be able to just have Mongo only return the user IDs: https://docs.mongodb.com/manual/tutorial/project-fields-from-query-results/

(Edit: I see you didn't actually write this code, feel free to leave it if you want.)

**Review by Xodarap — APPROVED** [review]
❤️ re-factoring plus testing PR's

:shipit:

**Pull request merged**